### ISSUE TITLE
Weight Decay configuration for BERT?

### ISSUE THREAD

**Issue by zphang** [nyu-mll/jiant#937]
The Transformers/pytorch_pretrained_bert version of the BERT training code applies weight decay to only a subset of the parameters: https://github.com/huggingface/transformers/blob/0856a231c0d6f3abb193e99ecf52ff76edb75fb8/examples/run_classifier.py#L482-L509

(I am referencing pytorch_pretrained_bert v0.6 because AllenNLP uses BertAdam from that version.)

As far as I can tell, we do not have the equivalent setup in jiant. I believe the current setup applies the default weight decay of 0.01 (which is the correct weight decay for the parameters that should have it applied) to all parameters.

Given that we have been able to train performant models with the current setup, I'm not sure if this qualifies as a bug or if we are satisfied with this arrangement.

**Comment by W4ngatang** [comment]
:+1:

There's also been some changes to optimizers that we haven't kept up with, e.g. `BertAdam` has an option for `amsgrad` that defaults to `False`.

**Comment by zphang** [comment]
*This is an automatically generated comment.*

As we update `jiant` to v2.x, `jiant` v1.x has been migrated to https://github.com/nyu-mll/jiant-v1-legacy. As such, we are closing all issues relating to `jiant` v1.x in this repository.

If this issue is still affecting you in `jiant` v1.x, please follow up at https://github.com/nyu-mll/jiant-v1-legacy/issues/937.

If this issue is still affecting you in `jiant` v2.x, reopen this issue or create a new one.
